codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -479,4 +274,158 @@ public HttpHeaders clear() {
         super.clear();
         return this;
     }
+
+    @Override
+    public int hashCode() {
+        return size();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof DefaultHttpHeaders)) {
+            return false;
+        }
+        DefaultHttpHeaders headers = (DefaultHttpHeaders) other;
+        if (size() != headers.size()) {
+            return false;
+        }
+        Comparator<CharSequence> valueComparator = AsciiString.CHARSEQUENCE_CASE_SENSITIVE_ORDER;
```

Comment:
@Scottmitch well not in `DefaultHeaders` as it doesn't know about `Comparators`.